devito/ir/iet/visitors.py

```diff
@@ -254,23 +255,30 @@ def visit_AugmentedExpression(self, o):
 
     def visit_LocalExpression(self, o):
         if o.write.is_Array:
+            ctype = [o.expr.lhs._C_typedata]
+            if o.write.volatile:
+                ctype.insert(0, 'volatile')
+            ctype = ' '.join(ctype)
             lhs = '%s%s' % (
                 o.expr.lhs.name,
                 ''.join(['[%s]' % d.symbolic_size for d in o.expr.lhs.dimensions])
             )
         else:
+            ctype = o.expr.lhs._C_typedata
             lhs = ccode(o.expr.lhs, dtype=o.dtype)
 
-        return c.Initializer(c.Value(o.expr.lhs._C_typedata, lhs),
-                             ccode(o.expr.rhs, dtype=o.dtype))
+        return c.Initializer(c.Value(ctype, lhs), ccode(o.expr.rhs, dtype=o.dtype))
 
     def visit_ForeignExpression(self, o):
         return c.Statement(ccode(o.expr))
 
-    def visit_Call(self, o):
+    def visit_Call(self, o, nested_call=False):
```

Comment:
I should review the code in more detail, but is there a quick and easy explanation as to why `nested_call` is needed?

Reply:
I need to pass it to `MultilineCall`, IIRC to distinguish between Call statements (terminating with ';') and Call expressions (no ';', thus making it possible to nest Calls)